src/JuMP.jl

```diff
@@ -439,6 +439,15 @@ end
 set_optimize_hook(model::Model, f) = (model.optimize_hook = f)
 
 
+"""
+    solve_time(model::Model)
+
+If available, returns the solve time reported by the solver.
```

Comment:
What happens if the solve time is not available? This is useful to document.

Reply:
Indeed, today it returns 
```julia
ERROR: ArgumentError: ModelLike of type ProxSDP.Optimizer does not support accessing the attribute MathOptInterface.SolveTime()
Stacktrace:
 [1] get(::ProxSDP.Optimizer, ::MathOptInterface.SolveTime) at /Users/guilhermebodin/.julia/packages/MathOptInterface/C3lip/src/attributes.jl:203
 [2] get(::MathOptInterface.Utilities.CachingOptimizer{ProxSDP.Optimizer,MathOptInterface.Utilities.UniversalFallback{JuMP._MOIModel{Float64}}}, ::MathOptInterface.SolveTime) at /Users/guilhermebodin/.julia/packages/MathOptInterface/C3lip/src/Utilities/cachingoptimizer.jl:438
 [3] get(::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.CachingOptimizer{ProxSDP.Optimizer,MathOptInterface.Utilities.UniversalFallback{JuMP._MOIModel{Float64}}},MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Bridges.AllBridgedConstraints{Float64}}}, ::MathOptInterface.SolveTime) at /Users/guilhermebodin/.julia/packages/MathOptInterface/C3lip/src/Bridges/bridgeoptimizer.jl:189
 [4] get(::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer,MathOptInterface.Utilities.UniversalFallback{JuMP._MOIModel{Float64}}}, ::MathOptInterface.SolveTime) at /Users/guilhermebodin/.julia/packages/MathOptInterface/C3lip/src/Utilities/cachingoptimizer.jl:438
 [5] _moi_get_result(::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer,MathOptInterface.Utilities.UniversalFallback{JuMP._MOIModel{Float64}}}, ::MathOptInterface.SolveTime) at /Users/guilhermebodin/Documents/JuMP.jl/src/JuMP.jl:596
 [6] get(::Model, ::MathOptInterface.SolveTime) at /Users/guilhermebodin/Documents/JuMP.jl/src/JuMP.jl:609
 [7] solve_time(::Model) at /Users/guilhermebodin/Documents/JuMP.jl/src/JuMP.jl:448
 [8] top-level scope at none:0
```

but we could redesign as 

```julia
function _try_get_solve_time(model_like)
    try
        return MOI.get(model_like, MOI.SolveTime())
    catch ex
        if isa(ex, ArgumentError)
            return "SolveTime() attribute not implemented by the optimizer."
        else
            rethrow(ex)
        end
    end
end

"""
    solve_time(model::Model)

If available, returns the solve time reported by the solver.
Returns "SolveTime() attribute not implemented by the optimizer." if the attribute is
not implemented.
"""
function solve_time(model::Model)
    return _try_get_solve_time(model)
end
```
@blegat

Reply:
I don't think 9e40df38ee3367e5097a738b31a1b39d702ff0ac is needed. Catching all ArumentError is too wide and the error message is already quite clear. Adding a sentence in the docstring is enough